subprojects/composite-builds/src/main/java/org/gradle/composite/internal/ErrorHandlingIncludedBuildExecuter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.composite.internal;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.component.BuildIdentifier;
+import org.gradle.initialization.IncludedBuildExecuter;
+import org.gradle.initialization.ReportedException;
+import org.gradle.internal.exceptions.Contextual;
+import org.gradle.internal.exceptions.LocationAwareException;
+
+class ErrorHandlingIncludedBuildExecuter implements IncludedBuildExecuter {
+    private final IncludedBuildExecuter delegate;
+
+    public ErrorHandlingIncludedBuildExecuter(IncludedBuildExecuter delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void execute(final BuildIdentifier targetBuild, final Iterable<String> taskNames) {
+        try {
+            delegate.execute(targetBuild, taskNames);
+        } catch (ReportedException e) {
+            throw contextualizeFailure(targetBuild, e);
+        }
+    }
+
+    private RuntimeException contextualizeFailure(BuildIdentifier buildId, ReportedException e) {
```

Comment:
I'm actually thinking that it might be better to remove the exception wrapping altogether. So instead of:
```
* What went wrong:
Failed to execute tasks for build 'buildB'
> Execution failed for task ':buildB:jar'.
   > jar task failed
```
we'd get:
```
* What went wrong:
Execution failed for task ':buildB:jar'.
> jar task failed
```

And instead of:
```
* What went wrong:
Failed to execute tasks for build 'buildB'
> Failed to execute tasks for build 'buildC'
   > Could not determine the dependencies of task ':buildC:compileJava'.
      > Included build dependency cycle: build 'buildB' -> build 'buildC' -> build 'buildB'
```
we'd have:
```
* What went wrong:
Could not determine the dependencies of task ':buildC:compileJava'.
> Included build dependency cycle: build 'buildB' -> build 'buildC' -> build 'buildB'
```

@big-guy WDYR? (Being Australian, the "R" is for "Reckon")